depocs/__init__.py

```diff
@@ -238,9 +239,13 @@ def open(self, call_site_level=1):
         self._Scoped__is_open = True
         self._Scoped__is_used = True
 
-        stack = inspect.stack()
-        if len(stack) > call_site_level:
-            self._Scoped__open_site = stack[call_site_level]
+        try:
+            frame = sys._getframe(call_site_level)
+        except ValueError:
+            # No frame found, skip
+            pass
+        else:
+            self._Scoped__open_site_frame = frame
```

Comment:
Any reason we're not just directly assigning this in line 243? 

Not a huge deal either way I'm just curious if there's a reason we made it a 2-liner that necessitated the `else` clause.

Reply:
No particular reason anymore, i'll simplify this try-except